test/axon/loop_test.exs

```diff
@@ -787,6 +787,29 @@ defmodule Axon.LoopTest do
                File.ls!("checkpoint") |> Enum.sort()
     end
 
+    test "saves a checkpoint on custom events", %{loop: loop} do
+      data =
+        List.duplicate({Nx.iota({1, 1}), Nx.iota({1, 1})}, 5)
+
+      assert %Axon.Loop.State{epoch: 3, iteration: 0, event_counts: %{iteration_completed: 15}} =
+               loop
+               |> Map.put(:output_transform, & &1)
+               |> Loop.checkpoint(event: :iteration_completed, filter: [every: 2])
+               |> Loop.run(data, %{}, epochs: 3)
+
+      assert [
+               "checkpoint_0_0.ckpt",
+               "checkpoint_0_2.ckpt",
+               "checkpoint_0_4.ckpt",
+               "checkpoint_1_1.ckpt",
+               "checkpoint_1_3.ckpt",
+               "checkpoint_2_0.ckpt",
```

Comment:
@seanmor5 looks like checkpointing does work (probably I did something wrong on my end). Though I didn't expect the event count (and thus the filter) for iterations to be a total tally over the whole run. Is this expected?

Reply:
Right. I can see the argument for both. Just wanted to make sure this is the expected format.